### PR TITLE
[Feature] Localization API addition for layoutDirection with countrycode & language combination

### PR DESCRIPTION
**Pull request by Mohtasim**
## Purpose

* Adding layoutDirection parameter with language & countryCode parameter

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What to Check
Verify that the following are valid
* Check contractor options from demo app launcher

### CODE REVIEW

File: azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/configuration/LocalizationConfiguration.java [hunk 1]
```diff
@@ -34,6 +36,15 @@ public final class LocalizationConfiguration {
     private int layoutDirection;
     private final Locale locale;
 
+    /**
+     * Create Localization configuration.
+     *
+     * @param locale The {@link Locale}; eg,. Locale.US
+     */
+    public LocalizationConfiguration(final Locale locale) {
```
Comment (pavelprystinka) [inline azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/configuration/LocalizationConfiguration.java:44]: This constructor does not have NonNull

File: azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/CallCompositeActivity.kt [hunk 2]
```diff
@@ -153,7 +153,7 @@ internal class CallCompositeActivity : AppCompatActivity() {
                 supportedOSLocale()
             }
             else -> {
-                configuration.localizationConfig!!.layoutDirection.let {
+                configuration.localizationConfig!!.layoutDirection?.let {
```
Comment (pavelprystinka) [inline azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/CallCompositeActivity.kt:156]: is layoutDirection nullable?

**Review by pavelprystinka — CHANGES_REQUESTED** [review]

File: azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/configuration/LocalizationConfiguration.java (hunk 1 above)
Comment (Mohtasim) [reply azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/configuration/LocalizationConfiguration.java]: Updated with Nonnull annotation.

File: azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/CallCompositeActivity.kt (hunk 2 above)
Comment (Mohtasim) [reply azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/presentation/CallCompositeActivity.kt:156]: No. As we are making it final and assigned value where we are not taking the direction from parameter. removing the nullable operator.

File: azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/configuration/LocalizationConfiguration.java [hunk 3]
```diff
@@ -31,16 +35,26 @@
...
     private final Locale locale;
 
+    /**
+     * Create Localization configuration.
+     *
+     * @param locale The {@link Locale}; eg,. Locale.US
+     */
+    public LocalizationConfiguration(@NonNull final Locale locale) {
+        this.locale = locale;
+        this.layoutDirection = LayoutDirection.LTR;
```
Comment (pavelprystinka) [inline azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/configuration/LocalizationConfiguration.java:48]: As we decided to make layoutDirection nullable, we do not need to set default value. Same for line 69, 91

**Review by pavelprystinka — CHANGES_REQUESTED** [review]

File: azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/configuration/LocalizationConfiguration.java (hunk 3 above)
Comment (Mohtasim) [reply azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/configuration/LocalizationConfiguration.java]: Got it. Also updated the java class links as well.

**Review by pavelprystinka — DISMISSED** [review]

**Review by pavelprystinka — APPROVED** [review]

**Review by iaulakh — APPROVED** [review]

**Pull request merged**